docs/guide/widgets-and-templates/fragments.md

```diff
@@ -0,0 +1,148 @@
+# Fragments are the new macros
+
+[Nunjucks macros](https://mozilla.github.io/nunjucks/templating.html#macro) are great tools for sharing template code across files. They have one significant limitation, however, in that they run synchronously. If you include template tags in them that run asynchronously, those tags will seem to not run at all when it comes to rendering.
+
+This poses a challenge when using the new [async components](/guide/async-components.md/async-components.md) in Apostrophe 3. In addition, the [area syntax changed](/guide/major-changes.md#areas-and-pages) to support those async components, so areas won't work in synchronous macros either. To enable developers to continue writing maintainable, [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) code, A3 has a new feature, **fragments**.
+
+## Macros are still supported
+
+Before getting into fragment's syntax, it is important to make one thing clear right away:
+
+**Macros aren't disabled or deprecated in A3.** Macros are still useful, and there is no great reason to force legacy projects to refactor macros that do not include areas already. Apostrophe developers can still use macros as they have been for simple and synchronous use cases.
+
+As you'll see, fragment syntax intentionally mimics macro syntax. It shouldn't be too difficult to refactor macros as fragments if you choose. In fact, **the current best practice recommendation is to only use fragments**. The primary reason for this is simply so that you only need to remember one syntax. Fragments can do ([nearly](#fragment-limitations)) everything that macros can do, so it might be simpler to use that one feature rather than have to think each time which one you should use.
+
+## Using fragments
+
+With that out of the way, how do they work?
+
+Fragments are defined with the `{% fragment %}` template tag, followed by a name for the fragment. The fragment closes with the `{% endfragment %}` tag.
+
+```django
+{% fragment heading() %}
+  <h2>This is a heading fragment</h2>
+{% endfragment %}
+```
+
+One difference from macros is that they are then called using the `render` template tag:
+
+```django
+{% render heading() %}
+{# `<h2>This is a heading fragment</h2>` #}
```

Comment:
I'm not sure the comment syntax is a good way to express this, maybe a separate markup code block showing the output.

Reply:
Improved to `{# Renders: `<h2>This is a heading fragment</h2>` #}`?